main.py

```diff
@@ -187,11 +187,20 @@ async def main():
     if args.mode == 'trade':
         if not args.config:
             parser.error('--config is required when mode is "trade"')
+
+        # Check if the config exists before going into the loop 
+        try:
+            parse_config(args.config)
```

Comment:
I like this as a first step! We could add some validation here potentially:

https://github.com/r0fls/soad/blob/main/utils/config.py#L113-L116

E.g. raise an error if something is missing here that we think should be required. However since this is a single entry point across each microservice (somewhat regrettably), bear in mind the config could be somewhat dynamic at this point.